site/src/pages/components/popover-hint.research.explainer.mdx

```diff
@@ -0,0 +1,195 @@
+
+To facilitate these delays, two new CSS properties are added:
+
+- `popover-show-delay` - applied to the invoking element. The invoking element must be hovered, focused, or long-pressed for this long before the popover is shown. The default value for this property is `0.5s`.
+- `popover-hide-delay` - applied to the popover. The invoking element *or* the popover must be **not** hovered, focused, or long-pressed for this long before the popover is hidden. The default value for this property is `infinity`.
+
+It is important to "debounce" these delays. Said another way, the delays are interpreted as *the time an element must be continuously hovered or de-hovered*. The naive Javascript implementation of this behavior is to just call `setTimeout(() => show/hidePopover(), delayValue)`. I.e. after the invoking element is hovered, wait `delayValue`, and then show the popover. But that does not properly handle the situation where the user hovers the element, and then de-hovers it, all within the delay time. In that case, no popover should be shown. Similarly, once the popover is shown and the user de-hovers the popover briefly but re-hoveres it again quickly, the popover should not close.
+
+## Example buttons
+
+Let's examine a small example:
+
+```html
+<button popovertarget=hint1 popovertargetaction=hover>
+  Button 1
+</button>
+<div popover=hint id="hint1">Hint 1</div>
+<button>Button 2</button>
+
+<style>
+  button    { popover-show-delay: 0.3s; }
+  [popover] { popover-hide-delay: 0.6s; }
+</style>
+```
+
+In this example, because of the `popover-show-delay: 0.3s`, hovering (with the mouse) or keyboard-focusing "Button 1", *and then waiting for 0.3 seconds* will cause "Hint 1" to be shown. It will stay showing as long as the button *or* the popover are hovered with the mouse or keyboard-focused.
+
+Because of the `popover-hide-delay: 0.6s`, the hint will be automatically hidden after 0.6 seconds, once "interest has left" the popover. That depends on how the popover was triggered:
+
+- If the invoker was **hovered** to trigger "Hint 1", then it will stay open as long as the invoker or the hint are still hovered. If the mouse if moved off of *both* the invoker and the hint for 0.6 seconds, the hint will be closed.
+- If the invoker was **keyboard focused** to trigger "Hint 1", then it will stay open as long as focus is contained within either the invoker or the hint. If keyboard focus leaves both invoker and hint for 0.6 seconds, the hint will be closed.
+
+## Popover types
+
+The new hover-triggering functionality should work for all `popover` types. There are use cases, e.g. for `popover=auto`; for example, a nested menu structure where sub-menus are shown when items in the primary menu are hovered or focused.
+
+## Example menu
+
+```html
+<button popovertarget=menu1>Menu</button>
+<div role=menu popover id=menu1>
```

Comment:
if this changes to role=menu, then there's a lot of extra updates that need to happen to make the markup pattern correct (role=none on all the li elements, role=menuitem on all the buttons, maybe even use tabindex=-1 to allude to the arrow key navigation)

but if we change this to a `ul`, then it's just a list like it was with the `menu` element, which means this would all just be tab key navigation and we don't need to distract with all the extra ARIA/implied keyboard functionality.  

we can go either way, it's fine.  we just need to add in all the other attributes if we go this route.

Reply:
Ok, thanks for the suggestion. I've changed this to `<ul>`s, and removed the `role`.